core/src/main/scala/com/pingcap/tispark/TiBatchWrite.scala

```diff
@@ -570,6 +498,190 @@ class TiBatchWrite(@transient val df: DataFrame,
 
     TableCodec.encodeRow(tiTableInfo.getColumns, convertedValues, tiTableInfo.isPkHandle)
   }
+
+  private def generateRDDToBeInserted(
+    rdd: RDD[ToBeCheckedRow]
+  ) = {
+    var allocatedRdd: RDD[(ToBeCheckedRow, Long)] = null
+    if (!tiTableInfo.isPkHandle) {
+      val step = rdd.count
+      val catalog = TiSessionCache.getSession(tiConf).getCatalog
+      val start = RowIDAllocator
```

Comment:
can you add a comment what this `start` means?

Reply:
done

Reply:
done